src/tentaclio/clients/databricks_client.py

```diff
@@ -0,0 +1,99 @@
+"""Databricks query client.
+
+Extends SQLAlchemyClient adding methods to get simba driver and http_path.
+"""
+
+import os
+import platform
+import urllib
+from typing import Union
+
+from sqlalchemy.engine import Connection, create_engine, result
+
+from tentaclio import urls
+
+from . import decorators, sqla_client
+
+
+__all__ = ["DatabricksClient"]
+
+SIMBA_DRIVER_DEFAULT_PATHS = {
+    "Windows": "path/to/simba/",
+    "Darwin": "/Library/simba/spark/lib/libsparkodbc_sbu.dylib",
+    "Linux": "/opt/simba/spark/lib/64/libsparkodbc_sb64.so",
+}
+
+
+class DatabricksClient(sqla_client.SQLAlchemyClient):
+    """Databricks client, backed by a pyodbc + SQLAlchemy connection."""
+
+    def __init__(self, url: Union[str, urls.URL]) -> None:
+        """Create sqlalchemy client based on the passed url.
+
+        This is a wrapper for sqlalchemy engine/connection creation.
+        """
+        super().__init__(url)
+        self.driver = _get_simba_driver_path()
+        self.http_path = self._get_http_path()
+
+    def _get_http_path(self) -> str:
+        http_path = os.environ.get("TENTACLIO__DATABRICKS_HTTP_PATH", "")
+        if http_path == "" and self.url.query:
+            http_path = self.url.query["http_path"]
+        return http_path
+
+    def _connect(self) -> Connection:
+        connection_url = _build_odbc_connection_string(
+            DRIVER=self.driver,
+            HOST=self.host,
+            PORT=self.port,
+            UID=self.username,
+            PWD=self.password,
+            Schema=self.database,
+            HTTPPath=self.http_path,
+            AuthMech=3,
+            SparkServerType=3,
+            ThriftTransport=2,
+            SSL=1,
+            IgnoreTransactions=1,
+        )
+
+        if self.engine is None:
+            self.engine = create_engine(
+                f"mssql+pyodbc:///?odbc_connect={connection_url}"
+            )
+        return self.engine.connect()
+
+    # Query methods:
+
+    @decorators.check_conn
+    def query(self, sql_query: str, **kwargs) -> result.ResultProxy:
+        """Execute a read-only SQL query, and return results.
+
+        This will not commit any changes to the database.
+        """
+        return self.conn.execute(sql_query, **kwargs)
+
+    @decorators.check_conn
+    def execute(self, sql_query: str, **kwargs) -> None:
+        """Execute a raw SQL query command."""
+        trans = self.conn.begin()
+        try:
+            self.conn.execute(sql_query, **kwargs)
+        except Exception:
+            trans.rollback()
+            raise
+        else:
+            trans.commit()
```

Comment:
i think that this solution is functionally the same as not overwriting the method anyway - `params` could still be set (this time as a `kwarg` rather than as its own named variable) and would still be passed to `self.conn.execute`, and then sqlalchemy would fail

f we're just stripping the `params`, could we just do `return super().executre(sql_query, params=None, **kwargs)` to solve this issue, and not have to repeat the logic? And perhaps could just add a logger line that we're stripped it if it's called with a `params` that isn't `None`.

Reply:
I did try this but that will still pass [parameters: {'params': None}] to sqlalc which causes pyodbc to complain. Will modify the sqlalc parent class directly and see how that looks

Reply:
ahh right - in that case your original might be better, or we could just remove the parent method `params` param entirely and leave it to the `**kwargs`? though that's probably risking a chesterton's fence problem so we should understand why it's explicitly named in the first place first

Reply:
That sounds nice. Yeah I'm not a huge fan of the `pass_params` approach, dropping params as you suggest sounds like an elegant approach. Let's see if @capitancambio has any strong opinions